backend/app/controllers/application_controller.rb

```diff
@@ -1,4 +1,19 @@
 # frozen_string_literal: true
 
 class ApplicationController < ActionController::API
+  def authorize_request
+    header = request.headers["Authorization"]
+    token = header.split(" ").last if header
+
+    begin
+      decoded = JwtService.decode(token)
+      @user = User.find(decoded[:user_id])
+    rescue ActiveRecord::RecordNotFound
+      render(json: { errors: ["User does not exists"] }, status: :unauthorized)
+    rescue JWT::DecodeError
+      render(json: { errors: ["Invalid token provided"] }, status: :unauthorized)
```

Comment:
I would suggest making a blanket error message. 

Then you can group rescue statements like so:

```
rescue ActiveRecord::RecordNotFound, JWT::DecodeError => e
      render(json: { errors: ["bla"] }, status: :unauthorized)
```

If you need to know what kind of error it is - I suggest logging the error deets manually or via an error monitoring tool. I leave that to you.